File: 백준/Silver/22869. 징검다리 건너기 （small）/징검다리 건너기 （small）.py
import sys

n, k = map(int, sys.stdin.readline().split())
my_list = list(map(int, sys.stdin.readline().split()))
d = [0]*n
d[0] = 1

for i in range(n):
    if d[i]:
        for j in range(i+1, n):
            if (j-i) * (1 + abs(my_list[i]-my_list[j])) <= k:
                d[j] = 1
                
if d[n-1]:
    print("YES")
else:
    print("NO")
            

File: 백준/Silver/22869. 징검다리 건너기 （small）/징검다리 건너기 （small）.cc
#include <iostream>
#include <vector>
#include <cmath>

using namespace std;

int main() {
    int n, k;
    cin >> n >> k;
    
    vector<int> my_list(n, 0);
    for (int i = 0; i < n; ++i) {
        cin >> my_list[i];
    }
    
    vector<int> d(n, 0);
    d[0] = 1;
    
    for (int i = 0; i < n-1; ++i) {
        if (d[i] == 1) {
            for (int j = i+1; j < n; ++j) {
                if (d[j] == 0 && (j-i)*(1+abs(my_list[i]-my_list[j])) <= k) {
                    d[j] = 1;
                }
            }
        }
    }
    
    if (d[n-1] == 1) {
        cout << "YES";
    }
    else {
        cout << "NO";
    }
    
    return 0;
}